scripts/core/menu/menu.ts

```diff
@@ -286,15 +289,26 @@ angular.module('superdesk.core.menu', [
 
                         _.each(scope.menu, (activity) => {
                             activity.isActive = route && route.href &&
-                                    route.href.substr(0, activity.href.length) === activity.href;
+                                route.href.substr(0, activity.href.length) === activity.href;
                         });
 
                         if (route && route.href) {
                             scope.activeMenuItemPath = getActiveMenuItemPath(route.href);
                         }
                     }
 
-                    scope.$on('$locationChangeStart', () => {
+                    switchAuthoring($location.absUrl());
+
+                    /**
+                     * Gets triggered after the location has been changed for the final time.
```

Comment:
`final time` is a bit confusing because the URL only changes one time to begin with. That other function that is being called multiple times is about the intention to change the URL, but can as well cancel the change. I think it might even be more clear without the comment because we have two listeners for different things which are named pretty well.